src/modules/job-exec/checkpoint.c

```diff
@@ -0,0 +1,261 @@
+/************************************************************\
+ * Copyright 2018 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* Prototype flux job exec service
+ *
+ * DESCRIPTION
+ *
+ * This module implements the exec interface as described in
+ * job-manager/start.c, but does not currently support execution of
```

Comment:
Header cut & pasted from job-exec.c.

A paragraph or two describing what this does (and its prototype nature) would be good actually!  (I got momentarily excited :-)